dev/src/bulk-writer.ts

```diff
@@ -436,6 +484,15 @@ export class BulkWriter {
     this.verifyNotClosed();
     const bulkCommitBatch = this.getEligibleBatch();
     const resultPromise = bulkCommitBatch.create(documentRef, data);
+    const pendingOp = bulkCommitBatch.getLastOp();
+    resultPromise.catch(err => {
+      const operation = new BulkWriterOperation(
+        documentRef,
+        'create',
+        pendingOp
+      );
+      this.errorFn(err, operation);
+    });
```

Comment:
Ideally, we would only attach our own Promise if `errorFn` is defined, but that would require users to register an `error` function before any other API call. I think that we could re-throw the original exception here though if the error callback is not defined. This can be a simple change if you change the default error function to just throw.

Reply:
Something like:

```
const defaultCatch = (e) => throw e;

set() {
	this.op().catch(() => defaultCatch())
}

onError(errorFn) {
  defaultCatch = errorFn;
}
```

This should avoid the "uncaught Promise rejection" handling only when `onError` is called.